### PR TITLE
SA-3142: JCPolicy Private Functions

### PR DESCRIPTION
**Pull request by gweinjc**
## Issues
* [SA-3142](https://jumpcloud.atlassian.net/browse/SA-3142) - JCPolicy Private Functions

## What does this solve?

- `Get-JCPolicyTemplateConfigField`: Returns an object array containing all available fields that can be edited for a given policy template
- `Set-JCPolicyConfigField`: Returns an object array containing fields that have had their values set for a given policy template
- `Show-JCPolicyValues`: Displays the current values for configurable fields for a given policy template and allows a user to specify what field they would like to edit; returns the index of the field

## Is there anything particularly tricky?
N/A

## How should this be tested?

```
$templateID = '5cf67c95232e11070490923f'

# Get configurable fields for the Windows - Device Installation Policy
$templateObject = Get-JCPolicyTemplateConfigField -templateID $templateID

# Display current values and prompt for user input on field they would like to edit
$fieldIndex = Show-JCPolicyValues -policyObject $templateObject

# Edit the selected field
$updatedPolicyObject = Set-JCPolicyConfigField -templateObject $templateObject -policyTemplateID $templateID -fieldIndex $fieldIndex

# Display policy object with newly set field and prompt for a new update
$fieldIndex = Show-JCPolicyValues -policyObject $templateObject -policyValues $updatedPolicyObject

# Edit the newly selected field
$updatedPolicyObject = Set-JCPolicyConfigField -templateObject $templateObject -policyTemplateID $templateID -policyValues $updatedPolicyObject -fieldIndex $fieldIndex

# Check new object
$updatedPolicyObject
```

[SA-3142]: https://jumpcloud.atlassian.net/browse/SA-3142?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: PowerShell/JumpCloud Module/Private/Policies/Set-JCPolicyConfigField.ps1 [hunk 1]
```diff
@@ -54,8 +56,8 @@ function Set-JCPolicyConfigField {
 
         #TODO: after showing the current values for the policy, prompt user if they want to update JUST one specific field by row:
```
Comment (jworkmanjc) [inline PowerShell/JumpCloud Module/Private/Policies/Set-JCPolicyConfigField.ps1:57]: Prompting the us is now done in `Show-JCPolicyValues`

**Review by jworkmanjc — APPROVED** [review]
Looks good to me, anything else I think we can address in the Set/New-Policy cards but AC/ functionality is met in this card

**Review by kmaranionjc — APPROVED** [review]

File: PowerShell/JumpCloud Module/Private/Policies/Enumerate-JCPolicyConfigMapping.ps1
[the remark this answers is not in the document]
Comment (gweinjc) [reply PowerShell/JumpCloud Module/Private/Policies/Enumerate-JCPolicyConfigMapping.ps1:1]: Changed

File: PowerShell/JumpCloud Module/Private/Policies/Set-JCPolicyConfigField.ps1 [hunk 2]
```diff
@@ -10,35 +10,37 @@ function Set-JCPolicyConfigField {
...
         $policyValues,
         # The policy name
-        [Parameter(Mandatory = $true)]
+        [Parameter(Mandatory = $false)]
         [System.String]
         $policyName,
         # The policy templateID
         [Parameter(Mandatory = $true)]
         [System.String]
         $policyTemplateID,
```
[the remark this answers is not in the document]
Comment (gweinjc) [reply PowerShell/JumpCloud Module/Private/Policies/Set-JCPolicyConfigField.ps1]: removed

**Pull request merged**